src/lib/persistence.ts

```diff
@@ -0,0 +1,184 @@
+import type { BaseTable } from "@/tables/types";
```

Comment:
We should add something table versioning to invalidate persistant values on new versions.

Reply:
Will add this, thanks.

So table versioning that removes all values on new version, and also automatically handle a schema diff, right?

`nextTables` represent the next session, when the world and tables are recreated and supposed to pickup previous values for persistent tables.

Reply:
Versioning or when it initially hydrates it will validate shape and empty if stored values do not conform.